Source/BugsnagDevice.m

```diff
@@ -7,7 +7,48 @@
 //
 
 #import "BugsnagDevice.h"
+#import "BugsnagConfiguration.h"
+#import "BugsnagCollections.h"
 
 @implementation BugsnagDevice
 
++ (BugsnagDevice *)deviceWithDictionary:(NSDictionary *)event {
+    BugsnagDevice *device = [BugsnagDevice new];
+    [self populateFields:device dictionary:event];
+    return device;
+}
+
++ (void)populateFields:(BugsnagDevice *)device
+            dictionary:(NSDictionary *)event {
+    NSDictionary *system = event[@"system"];
+    device.jailbroken = [system[@"jailbroken"] boolValue];
+    device.id = system[@"device_app_hash"];
+    device.locale = [[NSLocale currentLocale] localeIdentifier];
+    device.manufacturer = @"Apple";
+    device.model = system[@"machine"];
+    device.modelNumber = system[@"model"];
+    device.osName = system[@"system_name"];
+    device.osVersion = system[@"system_version"];
+    device.totalMemory = system[@"memory"][@"usable"];
+
+    NSMutableDictionary *runtimeVersions = [NSMutableDictionary new];
+    runtimeVersions[@"osBuild"] = system[@"os_version"];
+    runtimeVersions[@"clangVersion"] = system[@"clang_version"];
+    device.runtimeVersions = runtimeVersions;
+}
+
+- (NSDictionary *)toDict {
```

Comment:
Prior classes (`BugsnagConfiguration`, `BugsnagMetadata` etc.) use `toDictionary` as the method name.  I think we should be consistent.  And longer is more Cocoa.

Reply:
Renamed to `toDictionary`